src/Compilers/CSharp/Portable/Symbols/FileIdentifier.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+
+namespace Microsoft.CodeAnalysis.CSharp;
+
+internal struct FileIdentifier
+{
+    private static readonly Encoding s_encoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true);
+
+    public string? EncoderFallbackErrorMessage { get; init; }
+    public ImmutableArray<byte> FilePathChecksumOpt { get; init; }
+    public string DisplayFilePath { get; init; }
+
+    public static FileIdentifier Create(SyntaxTree tree)
+        => Create(tree.FilePath);
+
+    public static FileIdentifier Create(string filePath)
+    {
+        string? encoderFallbackErrorMessage = null;
+        ImmutableArray<byte> hash = default;
+        try
+        {
+            var encodedFilePath = s_encoding.GetBytes(filePath);
+            using var sha256 = SHA256.Create();
```

Comment:
Need to avoid hard coding any crypto algorithms in our code base. Instead need to use one of our centralized locations. When we don't centralize it means we get hassled hard as crypto algorithms are deprecated and we have to find all the individual places it was used. Likely should centralize in SourceHashAgorithms

Reply:
I wasn't sure what the correct usage would look like here. It seems like `SourceHashAlgorithms` itself just has helpers for converting to/from the GUID used to denote a hash algorithm in a portable PDB.